src/pages/home/report/LinkPreviewer.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {View, Image} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {uniqBy} from 'lodash';
+import Text from '../../../components/Text';
+import styles from '../../../styles/styles';
+import variables from '../../../styles/variables';
+import colors from '../../../styles/colors';
+
+const IMAGE_TYPES = ['jpg', 'jpeg', 'png'];
+
+const propTypes = {
+    /** Data about links provided in message. */
+    linkMetadata: PropTypes.arrayOf(
+        PropTypes.shape({
+            /**  The URL of the link. */
+            url: PropTypes.string,
+
+            /**  A description of the link. */
+            description: PropTypes.string,
+
+            /**  The title of the link. */
+            title: PropTypes.string,
+
+            /**  The publisher of the link. */
+            publisher: PropTypes.string,
+
+            /**  The image associated with the link. */
+            image: PropTypes.shape({
+                /**  The height of the image. */
+                height: PropTypes.number,
+
+                /**  The width of the image. */
+                width: PropTypes.number,
+
+                /**  The URL of the image. */
+                url: PropTypes.string,
+            }),
+
+            /**  The provider logo associated with the link. */
+            logo: PropTypes.shape({
+                /**  The height of the logo. */
+                height: PropTypes.number,
+
+                /**  The width of the logo. */
+                width: PropTypes.number,
+
+                /**  The URL of the logo. */
+                url: PropTypes.string,
+            }),
+        }),
+    ),
+
+    /** Maximum amount of visible link previews. -1 means unlimited amount of previews */
+    maxAmountOfPreviews: PropTypes.number,
+};
+
+const defaultProps = {
+    linkMetadata: [],
+    maxAmountOfPreviews: -1,
+};
+
+const LinkPreviewer = (props) =>
+    _.map(
+        _.take(uniqBy(props.linkMetadata, 'url'), props.maxAmountOfPreviews >= 0 ? Math.min(props.maxAmountOfPreviews, props.linkMetadata.length) : props.linkMetadata.length),
+        (linkData) => {
+            if (_.isArray(linkData)) {
+                return;
+            }
```

Comment:
I think so. I don't create this data myself just wait for data from backend :(

Reply:
Ah it's most likely because our stupid PHP API is turning empty objects into arrays when JSON encoding 😄  

I will make a note to fix this eventually. I think we can live with the array check for now so that this doesn't become a blocker.